### DISCUSSION TITLE
Static assets with Cloudflare Workers

### DISCUSSION THREAD

**Discussion by stephent** [honojs/hono#1465] [Q&A]
The [docs](https://hono.dev/getting-started/cloudflare-workers#serve-static-files) make reference to Workers Sites for this, but Cloudflare [states](https://developers.cloudflare.com/workers/configuration/sites/) 'Do not use Workers Sites for new projects.'

If I want to return a static asset (e.g. an image) from a Worker (*not* a Pages project) using Hono, what's the current recommended approach (assuming Workers Sites is not the best answer as of today)?

Thanks,

Stephen

**Comment by jeremiahlee** [comment]
I also ran into this today. I want to avoid Cloudflare Worker Sites since it is deprecated in favor of Cloudflare Pages. Cloudflare's and Hono's documentation recommend 2 different approaches and neither is ideal.

[Hono's documentation for Cloudflare Pages](https://hono.dev/getting-started/cloudflare-pages) is file-first, meaning it prioritizes static files over functions. All function requests must have the path prefix (eg `/api/*`). This is not ideal for API-centric use cases that happen to also need to serve a few files.

[Cloudflare Page's documentation for deploying a Hono site](https://developers.cloudflare.com/pages/framework-guides/deploy-a-hono-site/) is function-first, meaning it prioritizes function routes over static files. All static file requests must have a path prefix (eg `/public/*`) and those requests are handled by a Hono route for the static file directory forwarding the request to `ctx.env.ASSETS.fetch()`. The only way to serve a static file at the root is to manually define a Hono route for each static asset in the root directory, which is not ideal.

**Reply by jeremiahlee** [reply]
I looked more into the commit history of the Cloudflare documentation. It seems like the asset handling in Worker Sites will be overhauled soon to be more like Pages, so that's why they recommend using Pages for now. Unclear what the disruption will be from that "overhaul".

https://github.com/cloudflare/cloudflare-docs/issues/9739
